base/uk.ac.stfc.isis.ibex.ui.ioccontrol.tests/src/uk/ac/stfc/isis/ibex/ui/ioccontrol/table/tests/IOCPatternFilterTest.java

```diff
@@ -0,0 +1,68 @@
+
+/*
+* This file is part of the ISIS IBEX application.
+* Copyright (C) 2012-2022 Science & Technology Facilities Council.
+* All rights reserved.
+*
+* This program is distributed in the hope that it will be useful.
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v1.0 which accompanies this distribution.
+* EXCEPT AS EXPRESSLY SET FORTH IN THE ECLIPSE PUBLIC LICENSE V1.0, THE PROGRAM 
+* AND ACCOMPANYING MATERIALS ARE PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES 
+* OR CONDITIONS OF ANY KIND.  See the Eclipse Public License v1.0 for more details.
+*
+* You should have received a copy of the Eclipse Public License v1.0
+* along with this program; if not, you can obtain a copy from
+* https://www.eclipse.org/org/documents/epl-v10.php or 
+* http://opensource.org/licenses/eclipse-1.0.php
+*/
+
+package uk.ac.stfc.isis.ibex.ui.ioccontrol.table.tests;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import uk.ac.stfc.isis.ibex.configserver.IocState;
+import uk.ac.stfc.isis.ibex.ui.ioccontrol.table.IOCPatternFilter;
+
+@SuppressWarnings("checkstyle:methodname")
+@RunWith(MockitoJUnitRunner.class)
+public class IOCPatternFilterTest {
```

Comment:
In these tests we have only checked cases where we *do* have a match, i.e. the test would still pass if the code in `isLeafMatch` was just `return true;`. We need to test a case where we don't have a match, too.